### ISSUE TITLE
pjreddie_darknet trains, alexey_darknet does not

### ISSUE THREAD

**Issue by agirbau** [AlexeyAB/darknet#363]
Hello!
First of all, thank you for your work, it really helps a lot!
I've been lately working on detecting custom objects. The characteristics of one of the objects is that it is quite small, considering that the image is quite big (1920x1080). 
To detect it I used a custom net + random=0 to avoid resizing towards small images (e.g. 600x600) where the object disappears. 
The thing is that it trained quite well in the original branch of darknet, but it seems that using this one (with exactly the same configuration and dataset) doesn't train at all.
I've tried to train on bigger objects and it seems to work, so it may mean that does some resize somewhere that I haven't found. Do you have any clue about this? I'll keep checking though.

Again thank you for the work!

**Comment by AlexeyAB** [comment]
@agirbau 

* Can you show your cfg-file?
* On which cfg-file is your cfg-file based: yolo-voc.2.0.cfg, yolo-voc.cfg, tiny-yolo-voc.cfg?
* What neural network resolution did you use?

**Comment by agirbau** [comment]
Tomorrow I'll upload my .cfg. But answering the other 2 questions:

> On which cfg-file is your cfg-file based: yolo-voc.2.0.cfg, yolo-voc.cfg, tiny-yolo-voc.cfg? 

I don't exactly remember, one of the 2 yolo-voc though for sure. Do you think that this can be the issue for training in darknet_pjreddie and not this one?

> What neural network resolution did you use?

If you are asking for the input parameters "width" and "height", I downsampled the image to half, maintaining the division /32 ratio (in my case, as I do only 4 max-poolings I could put it to /16). The input resolution then is w = 980, h = 528. The final "grid" is 60x33.

**Comment by AlexeyAB** [comment]
1. Sometimes there are problems in training yolo-voc.cfg, but always training works well for yolo-voc.2.0.cfg.
2. Your input neural network resolution should be any value multiple of 16 in your case, but 980/16=**61,25**. So `width=980` is wrong parameter.

**Comment by agirbau** [comment]
Hello! 
Sorry for not answering until now.
I made a mistake on telling you 980, I meant to type 960, sry!
I copy here the .cfg file, it's a little custom net, I'll check later what you have said about yolo and yolov2 cfg files (I changed the # for -- for a correct visualization). Thank you! 

[net]
--Training
batch=5
subdivisions=5
width=960
height=528

channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 50400
policy=steps
steps=20000,25000
scales=.1,.1

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

---

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=64
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=12
activation=linear

[region]
-- anchors =  0.57273, 0.677385, 1.87446, 2.06253, 3.33843, 5.47434, 7.88282, 3.52778, 9.77052, 9.16828
-- anchors =  0.15, 0.15, 0.3, 0.3, 0.6, 0.6, 1, 1, 2, 2
anchors = 0.5762, 0.5762, 0.9743, 0.9743
bias_match=1
classes=1
coords=4
num=2
softmax=1
jitter=.2
rescore=0

object_scale=5
noobject_scale=1
class_scale=1
coord_scale=1

absolute=1
thresh = .6
random=0

**Comment by agirbau** [comment]
Hello!

I was finally able to train with the darknet_alexey. Is as you said, I was using yolo-voc.cfg as a baseline. The parameters that might cause this are (the ones I changed):

- learning_rate=0.001
+ learning_rate=0.0001
- burn_in=1000  <-- This one Idk what it does
- max_batches = 50400
+ max_batches = 45000
policy=steps
- steps=20000,25000
+ steps = 100, 20000, 35000
- scales=.1,.1
+ scales=10,.1,.1 

I'm guessing what is the difference in training between darknet_pjreddie and darknet_alexey, did you decide to change any policies for it?

Thank you, by the way!

**Comment by AlexeyAB** [comment]
@agirbau Hi,

Also I added `burn_in` as in pjreddie_darknet in the last commit: https://github.com/AlexeyAB/darknet/blob/64aa0180bb74e84a75958b3da0061a9f5615729d/src/network.c#L53

So you can try to update code from this repo and try to train model that based on `yolo-voc.cfg` instead of `yolo-voc.2.0.cfg`
